src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -671,11 +671,16 @@ grpc_error_handle Chttp2ServerListener::Create(
     // Create channelz node.
     if (grpc_channel_args_find_bool(args, GRPC_ARG_ENABLE_CHANNELZ,
                                     GRPC_ENABLE_CHANNELZ_DEFAULT)) {
-      std::string string_address = grpc_sockaddr_to_uri(addr);
+      auto string_address = grpc_sockaddr_to_uri(addr);
+      if (!string_address.ok()) {
+        return GRPC_ERROR_CREATE_FROM_CPP_STRING(
+            std::string(string_address.status().ToString()));
```

Comment:
No need to construct a new `std::string` here, since that's the type that `ToString()` already returns.

Reply:
Done.